fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubqueryToApply.java

```diff
@@ -73,7 +74,8 @@ public List<Rule> buildRules() {
                     LogicalFilter<Plan> filter = ctx.root;
 
                     ImmutableList<Set<SubqueryExpr>> subqueryExprsList = filter.getConjuncts().stream()
-                            .<Set<SubqueryExpr>>map(e -> e.collect(SubqueryExpr.class::isInstance))
+                            .<Set<SubqueryExpr>>map(e ->
+                                    e.collect(subq -> subq instanceof SubqueryExpr && !(subq instanceof ListQuery)))
```

Comment:
add some comments to explain why `e.collect(subq -> subq instanceof SubqueryExpr && !(subq instanceof ListQuery)))`